signer/src/transaction_coordinator.rs

```diff
@@ -535,6 +533,13 @@ where
 
         // Create a signal stream with the defined filter
         let signal_stream = self.context.as_signal_stream(presign_ack_filter);
+        // This value may not be the "correct" threshold to use. The
+        // threshold should be the max of all thresholds for the inputs in
+        // the sweep transaction package. Since the signer set is currently
+        // stable, using this value won't cause any issues. However we can
+        // have a bug here if we open up the signer set and allow a large
+        // increase in the signatures required parameter.
```

Comment:
We could also defensively bail with an informative message if a generated presign request happens to contain any inputs whose associated shares' `threshold` is >= the number of signers in the current signing set (shrunken set case), otherwise we'd just do unnecessary work and time out below with "not enough signatures" as `acknowledged_signers` would never be >= `signature_threshold`.

Shouldn't happen if packaging is fixed to properly account for differing thresholds, though (just saw https://github.com/stacks-sbtc/sbtc/issues/1767).

Reply:
Yeah, I figure that fix is better left for a separate PR, such as one that targets https://github.com/stacks-sbtc/sbtc/issues/1777

Reply:
Resolving, since we'll finalizing the handling of the threshold later.